src/python/pants/backend/terraform/target_types.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from pants.engine.rules import collect_rules
+from pants.engine.target import COMMON_TARGET_FIELDS, Dependencies, Sources, Target
+
+
+class TerraformSources(Sources):
+    expected_file_extensions = (".tf",)
+
+
+class TerraformModuleSources(TerraformSources):
+    default = ("*.tf",)
```

Comment:
Are there >1 terraform targets you could envision having who have `sources`? For example, I imagine there are no .tf test files.

If `TerraformModule` is the only one you expect to have a `Sources` field, then you can merge these into a single class for simplicity.

Reply:
There is an experimental `terraform test` command which can do unit testing of Terraform modules. https://www.terraform.io/docs/language/modules/testing-experiment.html. So I'm going to keep this code as-is for now.